python/essex-config/essex_config/doc_gen/doc_gen.py

```diff
@@ -23,8 +21,8 @@ def __find_subclasses(package_name: str) -> list[type]:
 
         # Iterate through all members of the module
         for _, obj in inspect.getmembers(module):
-            if inspect.isclass(obj) and issubclass(obj, Config) and obj is not Config:
-                subclasses.append(cast(type[Config[BaseModel]], obj))
```

Comment:
Nice! I would love if we start relying less and less in casting. In Python I've seen this more of a hack for the type checker than a good practice.